src/components/mdx/code-block.js

```diff
@@ -19,9 +19,10 @@ const codeBlockClassName = theme => {
 }
 
 const addNewlineToEmptyLine = line => {
-  if (line && line.length === 1 && line[0].empty) {
-    // Improves copy/paste behavior
-    line[0].content = "\n"
+  // addresses pasting bug on firefox by creating a new line at the end of a code block row
+  
+  if ( line && (line[line.length-1].content.trim() === "" || line[0].content.trim() === "")) {
+    line[line.length-1].content = line[line.length-1].content + " \n";
```

Comment:
In cases where we have very obtuse code like this, there should be a lengthy comment explaining just what the heck is going on. That or make the code more understandable